doc/appendices/command-line/traffic_ctl.en.rst

```diff
@@ -225,91 +213,113 @@ traffic_ctl server
    :program:`traffic_server` process.
 
 traffic_ctl plugin
 -------------------
 .. program:: traffic_ctl plugin
 .. option:: msg TAG DATA
 
-    Send a message to plugins. All plugins that have hooked the :cpp:enumerator:`TSLifecycleHookID::TS_LIFECYCLE_MSG_HOOK`
-    will receive a callback for that hook. The :arg:`TAG` and :arg:`DATA` will be available to the
-    plugin hook processing. It is expected that plugins will use :arg:`TAG` to select relevant messages
-    and determine the format of the :arg:`DATA`.
+   Send a message to plugins. All plugins that have hooked the
+   :cpp:enumerator:`TSLifecycleHookID::TS_LIFECYCLE_MSG_HOOK` will receive a callback for that hook.
+   The :arg:`TAG` and :arg:`DATA` will be available to the plugin hook processing. It is expected
+   that plugins will use :arg:`TAG` to select relevant messages and determine the format of the
+   :arg:`DATA`.
 
 traffic_ctl host
 ----------------
 .. program:: traffic_ctl host
+
+A stat to track status is created for each host. The name is the host fqdn with a prefix of
+"proxy.process.host_status". The value of the stat is a string which is the serialized
+representation of the status. This contains all of the current data ie, reasons, marked down times,
+and down time for each host.  The stats may be viewed using the :program:`traffic_ctl metric`
+command or through the `stats_over_http` endpoint.
+
+.. option:: --time count
+
+   Set the duration of an operation to ``count`` seconds. A value of ``0`` means no duration, the
+   condition persists until explicitly changed. The default is ``0`` if an operation requires a time
+   and none is provided by this option.
+
+.. option:: --reason active | local | manual
+
+   Sets the reason for the operation.
+
+   ``active``
+      Mark as if an active health check performed the operation.
+
+   ``local``
+      Mark as if a local health check performed the operation.
+
+   ``manual``
+      Done by direct administration command. This is the default is a reason is needed and not
+      provided by this option.
+
+   Internally the reason can be ``self_detect`` if
+   :ts:cv:`proxy.config.http.parent_proxy.self_detect` is set to the value 2 (the default) and
+   selection has determined the host is the local host. This is used to prevent parent selection
+   from creating a loop by selecting itself as the upstream.
+
 .. option:: status HOSTNAME [HOSTNAME ...]
 
-    Get the current status of the hosts used in parent.config as a next hop in a multi-tiered cache hierarchy.  The value 0 or 1 is returned indicating that the host is marked as down '0' or marked as up '1'.  If a host is marked as down, it will not be used as the next hop parent, another host marked as up will be chosen.
+   Get the current status of the hosts used in :file:`parent.config` as a next hop in a multi-tiered
+   cache hierarchy.  The value 0 or 1 is returned indicating that the host is marked as down '0' or
```

Comment:
The status is no-longer an int value but a string instead.  Given this csv string value for a host, 'HOST_STATUS_DOWN,ACTIVE:UP:0:0,LOCAL:UP:0:0,MANUAL:DOWN:1556896844:0,SELF_DETECT:UP:0" .
The overall host status is "HOST_STATUS_DOWN".  The remaining comma separated fields are the reason data.  In this example, The down reason is "MANUAL:DOWN" The two remaining colon separated fields in the manual reason are respectively the unix time stamp when the host was marked down, and finally the number of seconds the host is marked down for, 0 indicates indefinitely.  The SELF_DETECT reason is internal and does not have a down time field.  A self detected host is marked down indefinitely.